Tools/autotest/common.py

```diff
@@ -1318,16 +1318,20 @@ def test_arm_feature(self):
             self.set_parameter("DISARM_DELAY", 0)
         if self.mav.mav_type == mavutil.mavlink.MAV_TYPE_FIXED_WING:
             self.set_parameter("LAND_DISARMDELAY", 0)
-        # Rover and Sub don't have auto disarm
-        self.start_test("Test arm and disarm with switch")
-        arming_switch = 7
-        self.set_parameter("RC%d_OPTION" % arming_switch, 41)
-        self.set_rc(arming_switch, 1000)
-        if not self.arm_motors_with_switch(arming_switch):
-            raise NotAchievedException()
-        if not self.disarm_motors_with_switch(arming_switch):
-            raise NotAchievedException()
-        self.set_rc(arming_switch, 1000)
+        # Rover and Sub don't have auto disarm, Sub has no 'switches'
+        if self.mav.mav_type == mavutil.mavlink.MAV_TYPE_SUBMARINE:
+            if not self.disarm_vehicle():
+                self.progress("Failed to DISARM")
```

Comment:
match the other vehicles here.
```suggestion
                raise NotAchievedException()
```